src/app/tests/suites/TestLevelControlWithOnOffDependency.yaml

```diff
@@ -0,0 +1,225 @@
+
+config:
+    nodeId: 0x12344321
+    cluster: "Level Control"
+    endpoint: 1
+
+tests:
+    - label: "Wait for the commissioned device to be retrieved"
+      cluster: "DelayCommands"
+      command: "WaitForCommissionee"
+      arguments:
+          values:
+              - name: "nodeId"
+                value: nodeId
+
+    - label: "Sends a MoveToLevel command to set current level to min value"
+      command: "MoveToLevel"
+      arguments:
+          values:
+              - name: "level"
+                value: 1
+              - name: "transitionTime"
+                value: 0
+              - name: "optionMask"
+                value: 0
+              - name: "optionOverride"
+                value: 0
+
+    - label: "Wait 100 ms"
+      cluster: "DelayCommands"
+      command: "WaitForMs"
+      arguments:
+          values:
+              - name: "ms"
+                value: 100
+
+    - label: "Reads CurrentLevel attribute from DUT"
+      command: "readAttribute"
+      attribute: "current level"
+      response:
+          value: 1
+
+    - label: "Write OnOffTransitionTime attribute"
+      command: "writeAttribute"
+      attribute: "on off transition time"
+      arguments:
+          value: 0
+
+    - label: "Read OnOffTransitionTime attribute"
+      command: "readAttribute"
+      attribute: "on off transition time"
+      response:
+          value: 0
+
+    - label: "Write OnLevel attribute"
+      command: "writeAttribute"
+      attribute: "on level"
+      arguments:
+          value: 254
+
+    - label: "Read OnLevel attribute"
+      command: "readAttribute"
+      attribute: "on level"
+      response:
+          value: 254
+
+    - label: "Read MinValue attribute"
+      command: "readAttribute"
+      attribute: "on off transition time"
+      response:
+          value: 1
+
+    - label: "Send On Command"
+      cluster: "On/Off"
+      command: "On"
+
+    - label: "Check on/off attribute value is true after on command"
+      cluster: "On/Off"
+      command: "readAttribute"
+      attribute: "OnOff"
+      response:
+          value: 1
+          
+    - label: "Wait OnOffTransitionTime"
+      cluster: "DelayCommands"
+      command: "WaitForMs"
+      arguments:
+          values:
+              - name: "ms"
+                value: 100
+
+    - label: "If OnLevel is defined, check CurrentLevel is OnLevel value"
+      command: "readAttribute"
+      attribute: "current level"
+      response:
+          value: 254
+
+    - label: "Send Off Command"
+      cluster: "On/Off"
+      command: "Off"
+
+    - label: "Check on/off attribute value is false after off command"
+      cluster: "On/Off"
+      command: "readAttribute"
+      attribute: "OnOff"
+      response:
+          value: 0
+          
+    - label: "Wait OnOffTransitionTime"
+      cluster: "DelayCommands"
+      command: "WaitForMs"
+      arguments:
+          values:
+              - name: "ms"
+                value: 0
```

Comment:
if you wait for 0ms you don't need the delay at all

Reply:
I've just added the step in order to enhance that you should wait `OnOffTransitionTime`, as per specification. In my case, this attribute is disabled.